packages/paste-core/components/spinner/src/index.tsx

```diff
@@ -1,70 +1,50 @@
 import * as React from 'react';
-import * as PropTypes from 'prop-types';
-import {styled, keyframes, css, size as sizeFn} from '@twilio-paste/styling-library';
-import {LoadingIcon, LoadingIconProps} from '@twilio-paste/icons/esm/LoadingIcon';
-import {IconSize, isIconSizeTokenProp} from '@twilio-paste/style-props';
+import {useUID} from '@twilio-paste/uid-library';
+import {IconWrapper} from '@twilio-paste/icons/esm/helpers/IconWrapper';
+import type {IconWrapperProps} from '@twilio-paste/icons/esm/helpers/IconWrapper';
+import type {IconSize, TextColor} from '@twilio-paste/style-props';
+import {useTheme} from '@twilio-paste/theme';
 
-const rotate = keyframes`
-  0% {
-    transform: rotate(0deg);
-  }
-  100% {
-    transform: rotate(359deg);
-  }
-`;
+import {StyledCircleTrack, AnimatedStyledCircle, StyledSvg} from './styled';
 
-interface SpinnerWrapperProps {
-  size?: IconSize;
-  show: boolean;
-}
+type CircleGeometryProps = Pick<React.SVGProps<SVGCircleElement>, 'cx' | 'cy' | 'r'>;
 
-const SpinningWrapper = styled.div<SpinnerWrapperProps>(sizeFn, (props) =>
-  css({
-    display: 'inline-block',
-    opacity: props.show ? 1 : 0,
-    transition: 'opacity 250ms cubic-bezier(0.215, 0.61, 0.355, 1)',
-    animation: `${rotate} 2s linear infinite`,
-    '@media screen and (prefers-reduced-motion: reduce)': {
-      animation: 'none',
-    },
-  })
-);
+const circleGeometry: CircleGeometryProps = {
+  cx: 50,
+  cy: 50,
+  r: 45,
+};
 
-export interface SpinnerProps extends LoadingIconProps {
+export interface SpinnerProps extends IconWrapperProps {
   title: string;
   size?: IconSize;
   delay?: number;
+  decorative: boolean;
+  color?: TextColor;
 }
 
-const Spinner = React.forwardRef<HTMLDivElement, SpinnerProps>(
-  ({as, size, color, decorative, title, delay = 250}, ref) => {
+export const Spinner = React.forwardRef<HTMLDivElement, SpinnerProps>(
+  ({size = 'sizeIcon30', color = 'currentColor', title, as, display, decorative, delay = 250}, ref) => {
+    const titleId = `spinner-${useUID()}`;
+    const {borderWidths: {borderWidth40 = '8px'} = {}} = useTheme() || {};
```

Comment:
What's the concern around `useTheme()` returning null? I think we should assume it works and throw if it doesn't here.

Reply:
Today when I was running the unit tests, I noticed that if the theme provider is not used, `useTheme()` returns `undefined`. I don't know if we have consumers using our spinner, but not using theme, but I think it is possible.

Reply:
Pretty sure none of our components work without Theme.Provider visually. The tests that don't wrap it are so that the CSS values aren't modified from `space20` to the corresponding rem value, to make testing consistent even when tokens change.

Reply:
Totally. It's just that without the default values this causes the tests to throw an error.

Reply:
Are we confident that folks using our components are definitely using the theme? I'm hesitant to remove as I am concerned about breaking our downstream consumers' apps. If we are super confident it won't cause any issues  for consumers I can remove it.

Reply:
100% confident. Everything breaks without it, not just this component.